src/test/java/org/xwiki/git/script/GitScriptServiceTest.java

```diff
@@ -55,6 +56,54 @@
 
     private File testRepository;
 
+    @Unstable
+    private void getGitRepositoryAndFindAuthors(String username, String accessCode) throws Exception
+    {
+        GitScriptService service = this.componentManager.getInstance(ScriptService.class, "git");
+        Repository repository;
+        if (username == null && accessCode == null) {
+            repository = service.getRepository(this.testRepository.getAbsolutePath(), TEST_REPO_CLONED);
+        } else {
+            repository = service.getPrivateRepository(this.testRepository.getAbsolutePath(), TEST_REPO_CLONED,
```

Comment:
I think I would name the 2 APIs with the same method name and just change the parameters. Seems more consistent to me. `service.getRepository(... username, accessCode)`.

Reply:
This would also allow you use simply:

```
repository = service.getRepository(this.testRepository.getAbsolutePath(), TEST_REPO_CLONED, username, accessCode);
```

for both cases. And you'd check for null in the implementation class.

Reply:
You mean overloading methods as I was creating in the #2 ?

Reply:
Yes

Reply:
Doing it :)